src/ssl/support.cc

```diff
@@ -352,7 +352,7 @@ ssl_verify_cb(int ok, X509_STORE_CTX * ctx)
         if (error_no != SQUID_X509_V_ERR_INFINITE_VALIDATION) {
             if (check) {
                 ACLFilledChecklist *filledCheck = Filled(check);
-                assert(!filledCheck->sslErrors);
```

Comment:
I believe the removed assertion was conceptually wrong because this certificate validation code cannot know what has happened before it and should not assume that there were no errors earlier.

However, instead of just removing this assertion we could add a cache.log `WARNING` because if there were earlier errors then there is an increased chance that admin's `ssl_error` ACLs will work incorrectly here. Those ACLs could have been written under the assumption that those earlier errors would be detected here, and they will not be detected because this `sslproxy_cert_error` context must focus on detecting the _current_ validation error as we now document in squid.conf.documented.

After the fix in this PR, there are no known (to me) use cases where `sslErrors` would be set in this context, but there is no fundamental reason they cannot exist. Before this PR, we could rely on the assertion to alert us of their existence -- as byg 5211 did -- but if we just remove the assertion, we may not know whether this conflict between old errors and the new validation error does happen in the real world.

Please let me know if you think adding a `WARNING` is a good idea. Here is an incomplete sketch:

```C++
if (filledCheck->sslErrors)
    debugs(83, Important(65), "WARNING: Previously seen TLS errors present but will not be detected by sslproxy_cert_error");
```

I am worried that the above will not be enough to explain what is going on. On the other hand, if we never actually see that `WARNING` in the real world, then we will not have to deal with the "explanation" part of the problem; we can delay solving that (by providing a detailed explanation somewhere and referencing it from the error message) until we confirm that the conflict is still real.

P.S. This problem is one of the reasons we probably need to split `ssl_errors` into two as discussed [elsewhere](https://github.com/squid-cache/squid/pull/1044/files#r867287851).